linuxPackages.drbd: 9.2.8 -> 9.2.9

birkb opened the pull request:
## Description of changes

- [6.8 kernel support](https://github.com/LINBIT/drbd/releases/tag/drbd-9.2.9) added
- aligned to [Nix standard build environment](https://ryantm.github.io/nixpkgs/stdenv/stdenv/)
- test added
- redundant `installPhase` and `postFixup` definitions removed

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

Commit:
linuxPackages.drbd: 9.2.8 -> 9.2.9
- 6.8 kernel support added
- aligned to Nix standard build environment
- test added
- redundant installPhase and postFixup definitions removed

Signed-off-by: birkb <birk@batchworks.de>

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/linuxpackages-drbd-missing-system-map-file-skipping-depmod/44848/1

Commit:
whitespaces fixed

Signed-off-by: birkb <birk@batchworks.de>

Commit:
whitespaces fixed

Signed-off-by: birkb <birk@batchworks.de>

birkb committed:
Merge branch 'NixOS:master' into drbd-kernel-module

Commit:
Merge remote-tracking branch 'upstream/master' into drbd-kernel-module

Thesola10 commented on nixos/tests/all-tests.nix:
```diff
@@ -266,6 +266,7 @@ in {
   dovecot = handleTest ./dovecot.nix {};
   drawterm = discoverTests (import ./drawterm.nix);
   drbd = handleTest ./drbd.nix {};
+  drbd-driver = handleTest ./drbd-driver.nix {};
```
The `drbd` test above uses the `services.drbd.enable` option, which already enables the kernel module. This test is redundant.

https://github.com/NixOS/nixpkgs/blob/44072e24566c5bcc0b7aa9178a0104f4cfffab19/nixos/modules/services/network-filesystems/drbd.nix#L43

birkb replied:
The [drbd](https://github.com/birkb/nixpkgs/blob/drbd-kernel-module/nixos/tests/drbd.nix) test validates the userland utilities, but not the kernel driver. The in-tree driver is loaded to create the DRBD devices, but no special driver checks are done.

The [drbd-driver](https://github.com/birkb/nixpkgs/blob/drbd-kernel-module/nixos/tests/drbd-driver.nix) test loads the out-of-tree driver (provided by this package) and validates the version of the loaded driver.

I think we need both tests...

Thesola10 reviewed (changes requested):
Correct me if I'm wrong, but the following commits do not fit NixOS contributor guidelines. All commits should be squashed into the first one, or split according to their semantic effect.

![image](https://github.com/NixOS/nixpkgs/assets/7287268/7f63ecc3-b98d-44ea-8b7a-e27cace46f82)

astro commented on nixos/tests/drbd-driver.nix:
```diff
@@ -0,0 +1,19 @@
+
+  nodes = {
+    machine = { config, pkgs, ... }: {
+      boot.kernelModules = [ "drbd" ];
+      boot.extraModulePackages = with config.boot.kernelPackages; [ drbd ];
+      boot.kernelPackages = pkgs.linuxPackages;
```
```suggestion
      boot = {
        kernelModules = [ "drbd" ];
        extraModulePackages = with config.boot.kernelPackages; [ drbd ];
        kernelPackages = pkgs.linuxPackages;
      };
```

birkb replied:
Hi Astro, thanks for the hint. I have tweaked both areas.

astro commented on nixos/tests/drbd-driver.nix:
```diff
@@ -0,0 +1,19 @@
+import ./make-test-python.nix ({ lib, pkgs, ... }: {
+  name = "drbd-driver";
+  meta = with pkgs.lib.maintainers; {
+    maintainers = [ birkb ];
+  };
```
```suggestion
  meta.maintainers = with pkgs.lib.maintainers; [ birkb ];
```

Commit:
minor rewrite
- for better readability

Signed-off-by: birkb <birk@batchworks.de>

Commit:
linuxPackages.drbd: 9.2.8 -> 9.2.9
- 6.8 kernel support added
- aligned to Nix standard build environment
- test added
- redundant installPhase and postFixup definitions removed

Signed-off-by: birkb <birk@batchworks.de>

K900 commented:
Mass ping, please resubmit.
